docs/eeps/draft-maven-style-plugins.md

```diff
@@ -0,0 +1,237 @@
+---
+EEP: unnumbered
+Title: Maven-style Plugins
+Author: dmikurube
+Status: Draft
+Type: Standards Track
+Created: 2023-06-19
+---
+
+Introduction
+=============
+
+This EEP explains how Embulk's new Maven-style plugin mechanism is designed. Maven-style plugins are designed to overcome the difficulties of Embulk's old RubyGems-style plugins.
+
+Motivation
+===========
+
+Embulk's plugins have been distributed as RubyGems, even if they are coded in Java. The RubyGems-style plugins, however, have had several drawbacks.
+
+* RubyGems-style plugins require JRuby.
+    * JRuby needs a lot of resources, such as CPU time, Java heap memory, and metaspace.
+    * JRuby takes a long time to initialize itself.
+    * Making JRuby optional is another effort to be explained in another EEP. This EEP would compose a part of the effort.
+* Bundler is mandatory to select a specific version of the same RubyGems-style plugin if multiple versions are installed.
+    * Users sometimes wanted to select a specific version from the same installation set.
+    * Reconfiguring Bunlder needs time, and sometimes reinstallation.
+* A RubyGems-style plugin has to include all the dependency JAR files packaged.
+    * The size of a RubyGems-style plugin package tended to be large.
+    * Many Embulk plugins are expected to depend on the same Java libraries. They would be duplicated on the network for downloading, and on the file system.
+    * License matters are more complicated in a RubyGems-style plugin because it includes redistributions.
+
```

Comment:
The Maven-style plugin allows the same plugin name. The RubyGem-style plugin doesn't allow it. How about adding that description as motivation?